jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/XMLUtil.java

```diff
@@ -0,0 +1,88 @@
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import java.io.File;
+import java.io.IOException;
+
+public class XMLUtil {
+    private XMLUtil() { }
+
+    public static Document createNewDocument() throws ParserConfigurationException {
+        DocumentBuilderFactory documentBuilderFactory = getDocumentBuilderFactory();
+        return documentBuilderFactory.newDocumentBuilder().newDocument();
+    }
+
+    public static Document readXML(File xmlFile) throws ParserConfigurationException, IOException, SAXException {
+        DocumentBuilderFactory documentBuilderFactory = getDocumentBuilderFactory();
+        DocumentBuilder documentBuilder = documentBuilderFactory.newDocumentBuilder();
+
+        return documentBuilder.parse(xmlFile);
+    }
+
+    public static void writeXML(Document document, File xmlFile) throws TransformerException {
+        TransformerFactory transformerFactory = TransformerFactory.newInstance();
+        transformerFactory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+        transformerFactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_DTD, "");
+        transformerFactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_STYLESHEET, "");
+        Transformer transformer = transformerFactory.newTransformer();
+        document.setXmlStandalone(true);
+        DOMSource source = new DOMSource(document);
+        StreamResult result = new StreamResult(xmlFile);
+        transformer.transform(source, result);
+    }
+
+    public static Node getNodeFromDocument(Document doc, String xPathExpression) throws XPathExpressionException {
+        XPathFactory xPathfactory = XPathFactory.newInstance();
+        XPath xpath = xPathfactory.newXPath();
+        return (Node) xpath.compile(xPathExpression).evaluate(doc, XPathConstants.NODE);
+    }
+
+    public static String getNodeValueFromDocument(Document doc, String xPathExpression) throws XPathExpressionException {
+        XPathFactory xPathfactory = XPathFactory.newInstance();
+        XPath xpath = xPathfactory.newXPath();
+        return (String) xpath.compile(xPathExpression).evaluate(doc, XPathConstants.STRING);
+    }
+
+    private static DocumentBuilderFactory getDocumentBuilderFactory() throws ParserConfigurationException {
+        DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+        documentBuilderFactory.setAttribute(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+        documentBuilderFactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_DTD, "");
+        documentBuilderFactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_SCHEMA, "");
+        documentBuilderFactory.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
```

Comment:
These urls can be converted into an array of strings and then we can use a loop for setting them here..

better than hardcoding